packages/surveys/src/components/wrappers/StackedCardsContainer.tsx

```diff
@@ -0,0 +1,147 @@
+"use client";
+
+import { useEffect, useMemo, useRef, useState } from "preact/hooks";
+
+import { TProductStyling } from "@formbricks/types/product";
+import { TCardArrangementOptions } from "@formbricks/types/styling";
+import { TSurvey, TSurveyStyling } from "@formbricks/types/surveys";
+
+interface StackedCardsContainerProps {
+  cardArrangement: TCardArrangementOptions;
+  currentQuestionId: string;
+  survey: TSurvey;
+  getCardContent: (questionIdx: number, offset: number) => JSX.Element | undefined;
+  styling: TProductStyling | TSurveyStyling;
+}
+
+export const StackedCardsContainer = ({
+  cardArrangement,
+  currentQuestionId,
+  survey,
+  getCardContent,
+  styling,
+}: StackedCardsContainerProps) => {
+  const [hovered, setHovered] = useState(false);
+  const highlightBorderColor =
+    survey.styling?.highlightBorderColor?.light || styling.highlightBorderColor?.light;
+  const cardBorderColor = survey.styling?.cardBorderColor?.light || styling.cardBorderColor?.light;
+  const cardRefs = useRef<(HTMLDivElement | null)[]>([]);
+  const resizeObserver = useRef<ResizeObserver | null>(null);
+  const [cardHeight, setCardHeight] = useState("auto");
+
+  const cardIndexes = useMemo(() => {
+    let cardIndexTemp = survey.questions.map((_, index) => index);
+    if (survey.welcomeCard.enabled) {
+      cardIndexTemp.unshift(-1);
+    }
+    if (survey.thankYouCard.enabled) {
+      cardIndexTemp.push(survey.questions.length);
+    }
+    return cardIndexTemp;
+  }, [survey]);
+
+  const questionIdx = useMemo(() => {
+    if (currentQuestionId === "start") return survey.welcomeCard.enabled ? -1 : 0;
+    if (currentQuestionId === "end") return survey.thankYouCard.enabled ? survey.questions.length : 0;
+    return survey.questions.findIndex((question) => question.id === currentQuestionId);
+  }, [currentQuestionId, survey.welcomeCard.enabled, survey.thankYouCard.enabled]);
```

Comment:
please check the linter warning and either extend the deps array (ideal case) or add an ignore statement